core/src/com/mygdx/game/ecs/systems/CollisionSystem.java

```diff
@@ -0,0 +1,44 @@
+package com.mygdx.game.ecs.systems;
+
+import com.badlogic.ashley.core.ComponentMapper;
+import com.badlogic.ashley.core.Entity;
+import com.badlogic.ashley.core.Family;
+import com.badlogic.ashley.systems.IteratingSystem;
+import com.badlogic.ashley.utils.ImmutableArray;
+import com.badlogic.gdx.math.Intersector;
+import com.badlogic.gdx.math.Polygon;
+import com.mygdx.game.ecs.GameEngine;
+import com.mygdx.game.ecs.components.BotComponent;
+import com.mygdx.game.ecs.components.BulletComponent;
+import com.mygdx.game.ecs.components.CollisionComponent;
+import com.mygdx.game.ecs.components.PositionComponent;
+import com.mygdx.game.ecs.components.SpriteComponent;
+import com.mygdx.game.ecs.components.VelocityComponent;
+
+public class CollisionSystem extends IteratingSystem {
+    private ComponentMapper<SpriteComponent> spriteMapper;
+    private ComponentMapper<CollisionComponent> collisionMapper;
+
+    public CollisionSystem() {
+        super(Family.all(BotComponent.class, SpriteComponent.class, PositionComponent.class).get());
+        spriteMapper = ComponentMapper.getFor(SpriteComponent.class);
+    }
+
+    public void processEntity(Entity entity, float deltaTime) {
+        ImmutableArray<Entity> bullets = GameEngine.getInstance().getEntitiesFor(Family.all(BulletComponent.class).get());
+        for (Entity bullet : bullets) {
+            if(colliding(bullet, entity)) {
+                GameEngine.getInstance().removeEntity(entity);
+                GameEngine.getInstance().removeEntity(bullet);
+            }
+        }
+    }
```

Comment:
Så om jeg forstår CollisionSystem rett her nå så er det bare zombier vi sjekker kollisjon med? Det er jo strengt tatt det eneste nødvendige for oss, selv om det ikke gjør ting veldig modifiable for andre ting (f.eks. hvis vi vil legge inn environment må vi lage et nytt system, eller endre dette systemet for å gjøre at player kan kollidere med environment. Men vi har jo ikke dette, så tenker ikke det er super-viktig at vi gjør det). 

Men hvis jeg da skal sjekke om zombie kolliderer med player, så lager jeg bare en ny if-setning under din for-loop, og sjekker if(colliding(player, entity)) (etter å ha hentet player entity), og så håndterer jeg bare det under?

Reply:
Ja, det er alt du trenger å gjøre for å implementere player-zombie collision.

Jeg er enig i at det er litt lite modifiable, men jeg kjenner ikke helt til workloaden til processEntity. Å legge inn en if-setning i processEntity som sjekker om entity har BotComponent virket som en unødvendig ekstra workload, men det er sikkert ikke det tyngste å legge til.